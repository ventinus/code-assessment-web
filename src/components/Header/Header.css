.header {
  padding-bottom: 15px;
  border-bottom: 1px solid #d8d8d8; }
  @media (min-width: 768px) {
    .header {
      padding-bottom: 25px;
      display: flex;
      flex-direction: row;
      justify-content: space-between;
      align-items: center; } }
  @media (min-width: 1024px) {
    .header {
      padding-bottom: 30px; } }
