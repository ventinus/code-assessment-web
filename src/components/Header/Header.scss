@import '../../scss/variables/mixins.scss';
@import '../../scss/variables/variables.scss';

.header {
  padding-bottom: 15px;

  @include breakpoint(md) {
    padding-bottom: 25px;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
  }

  @include breakpoint(lg) {
    padding-bottom: 30px;
  }
}
