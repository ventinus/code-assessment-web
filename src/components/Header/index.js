import React from 'react'
import PropTypes from 'prop-types'

import { BorderSection } from '..'

import './Header.css'

const Header = ({ heading, children }) => (
  <BorderSection className="header">
    <h1 className="type--a1">{heading}</h1>
    {children}
  </BorderSection>
)

Header.propTypes = {
  heading: PropTypes.string.isRequired,
  children: PropTypes.any
}

export default Header
