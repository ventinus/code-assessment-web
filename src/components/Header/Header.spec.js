import React from 'react'
import renderer from 'react-test-renderer'
import Header from '.'

describe('Header component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header heading="The Heading">children</Header>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
