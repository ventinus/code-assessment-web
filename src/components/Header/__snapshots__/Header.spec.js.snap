// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<div
  className="border-section header"
  style={
    Object {
      "borderBottomWidth": 1,
    }
  }
>
  <h1
    className="type--a1"
  >
    The Heading
  </h1>
  children
</div>
`;
